---
title: "presentation_figures"
author: "Iris Kern"
date: "2023-10-03"
output: html_document
---


# Setup
```{r}
library(tidyverse)
library(ggplot2)
library(RColorBrewer)
library(plotly)
library(vegan)
library(heatmaply)
library(ggnewscale)

set.seed(20)
```

# Data
## Load in
```{r}
quant_data <- read_csv("csvs/quant_data.csv")
metadataframe <- read_csv("csvs/metadataframe.csv")

filter(quant_data, str_detect(metabolite, "Proline"))
```

## Wrangle
```{r}
data_vis <- quant_data %>%  
  mutate(treatment = str_extract(replicate, "Std|Poo|Blk|C|ZF|ZL|ZH|RL|RH|LL|LH|Tote")) %>% 
  mutate(timepoint = str_extract(replicate, "27June|30June|14July|21July|27July")) %>% 
  mutate(timepoint = str_replace_all(timepoint, "July", "-7-22")) %>% 
  mutate(timepoint = str_replace_all(timepoint, "June", "-6-22")) %>% 
  mutate(date = as.Date(timepoint, format = "%d-%m-%y")) %>% 
  select(metabolite, treatment, date, nM, replicate) %>% 
  filter(metabolite != "Sodium 2-mercaptoethanesulfonate") %>% 
  filter(metabolite != "Fumaric acid") %>% 
  na.omit()

level_order <- c("T0", "C", "ZF", "ZL", "ZH", "LL", "LH", "RL", "RH")

heat_level_order_treat <- c("2022-06-27 Tote", "2022-06-30 C",  "2022-07-14 C",  "2022-07-21 C",  "2022-07-27 C",  
                            "2022-06-30 ZF", "2022-07-14 ZF", "2022-07-21 ZF", "2022-07-27 ZF",
                            "2022-06-30 ZL", "2022-07-14 ZL", "2022-07-21 ZL", "2022-07-27 ZL", 
                            "2022-06-30 ZH", "2022-07-14 ZH", "2022-07-21 ZH", "2022-07-27 ZH",
                            "2022-06-30 LL", "2022-07-14 LL", "2022-07-21 LL", "2022-07-27 LL",
                            "2022-06-30 LH", "2022-07-14 LH", "2022-07-21 LH", "2022-07-27 LH", 
                            "2022-06-30 RL", "2022-07-14 RL", "2022-07-21 RL", "2022-07-27 RL",
                            "2022-06-30 RH", "2022-07-14 RH", "2022-07-21 RH", "2022-07-27 RH")

heat_level_order_date <- c("2022-06-27 Tote", "2022-06-30 C","2022-06-30 ZF", "2022-06-30 ZL", 
                           "2022-06-30 ZH", "2022-06-30 LL","2022-06-30 LH", "2022-06-30 RL", "2022-06-30 RH",
                           "2022-07-14 C", "2022-07-14 ZF", "2022-07-14 ZL", "2022-07-14 ZH", "2022-07-14 LL", 
                           "2022-07-14 LH", "2022-07-14 RL", "2022-07-14 RH", 
                           "2022-07-21 C", "2022-07-21 ZF", "2022-07-21 ZL", "2022-07-21 ZH", 
                           "2022-07-21 LL", "2022-07-21 LH", "2022-07-21 RL", "2022-07-21 RH", 
                           "2022-07-27 C", "2022-07-27 ZF", "2022-07-27 ZL",  
                           "2022-07-27 ZH", "2022-07-27 LL", "2022-07-27 LH", "2022-07-27 RL",  "2022-07-27 RH")
```

# heatmap
```{r}
quant_nmds_heat <- data_vis %>% 
  group_by(metabolite, treatment, date) %>%
  summarise(avg_nmol = mean(nM)) %>%
  ungroup() %>%
  mutate(id = paste(date, treatment)) 
# %>% 
#   filter(str_detect(treatment, "RH|RL|LH|LL"))

heatmap_data <- quant_nmds_heat %>% 
  # group_by(metabolite) %>%
  # filter(str_detect(treatment, "RH|RL|C|Tote")) %>% 
  # filter(avg_nmol >= 0.01) %>% 
  complete(metabolite, id, fill = list(avg_nmol = 0)) %>% 
  group_by(metabolite) %>% 
  # mutate(norm_conc = rank(avg_nmol)) %>%
  mutate(norm_conc = (avg_nmol - min(avg_nmol)) / (max(avg_nmol) - min(avg_nmol))) %>%
  select(metabolite, norm_conc, id) %>%
  pivot_wider(names_from = "metabolite", values_from = "norm_conc") %>%
  mutate(id=factor(id, levels=c(heat_level_order_treat))) %>%
  arrange(id) %>%
  column_to_rownames("id")


plotheatmap <- heatmap_data %>%
  data.matrix() %>% 
  heatmaply(fontsize_row = 10, fontsize_col = 7) # %>%
  # heatmaply(Rowv = FALSE, fontsize_row = 10, fontsize_col = 7)

# make clustered heatmap from scratch using pheatmap

plotheatmap
```

```{r}
ranked_conc <- quant_nmds_heat %>% 
  # group_by(metabolite) %>%
  # filter(str_detect(treatment, "RH|RL|C|Tote")) %>% 
  # filter(avg_nmol >= 0.01) %>% 
  complete(metabolite, id, fill = list(avg_nmol = 0)) %>% 
  group_by(metabolite) %>% 
  mutate(norm_conc = rank(avg_nmol)) %>%
  select(metabolite, norm_conc, id) 

data_vis_fc <- data_vis %>% 
  select(-c("replicate")) %>% 
  filter(!str_detect(treatment, "Tote")) %>% 
  pivot_wider(names_from = treatment, values_from = nM, values_fn = mean) %>% 
  pivot_longer(cols = c(ZF, ZL, ZH, LL, LH, RL, RH)) %>% 
  mutate(fold_change = value/C)

data_vis_km <- data_vis %>% 
  pivot_wider(names_from = metabolite, values_from = nM) %>% 
  select(-c("treatment", "date")) %>% 
  column_to_rownames("replicate") %>% 
  data.matrix() %>%
  `[<-`(is.na(.), 0) %>%
  scale() %>%
  t() %>%
  kmeans(centers = 5) %>% 
  pluck("cluster") %>%
  as.data.frame() %>%
  set_names("cluster") %>%
  rownames_to_column("metabolite")

data_vis_cluster <- data_vis_fc %>% 
  left_join(data_vis_km) %>% 
  rename(control = C, treatment = name, nM = value) %>% 
  mutate(cluster = factor(cluster)) %>% 
  filter(!str_detect(treatment, "Tote|ZF|ZL|ZH"))
```


```{r}
ranked_conc %>% 
  ggplot() +
  geom_raster(aes(x = metabolite, y = id, fill = norm_conc)) +
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) +
  scale_fill_gradient2(low = "steelblue", mid = "grey90",
                       high = "goldenrod1", midpoint = 16.5,
                       breaks=c(0, 16.5, 33), labels=c(
                         "Very little of a given compound",
                         "Median compound abundance",
                         "Compound maximum"
                       )) +
  new_scale_fill() + 
  geom_tile(aes(x=metabolite, y=-1.5, fill=cluster), height=3, data = data_vis_cluster) + 
  # scale_fill_discrete(guide="none") +
  # theme_void() +
  theme(legend.position = "top",
        legend.key.width = unit(dev.size()[1] / 6, "inches"),
        legend.text = element_text(size=8))
  # coord_cartesian(ylim=c(0, NA)) +
  # annotate("rect", xmin=-20, xmax=0, ymin=c(0.5, 4.5, 8.5, 12.5), ymax=c(4.5, 8.5, 12.5, 16.5),
  #          fill=c("#0d374f", "#367776", "#1f92d0", "lightblue2")) 
  # annotate("text", x = -11, y=c(2.5, 6.5, 10.5, 14.5), label=c("6-30", "7-14", "7-21", "7-27"),
  #          angle=90, color="white", size=6) +
  # annotate("text", x=205, y=-10, label="Metabolite", size=6) +
  # annotate("text", x=Inf, y=-10, label="k-means clusters with n=5", hjust=1) +
  # annotate("tile", x=312, y=seq(-8, -13, length.out=5), width=10, height=2,
  #          fill=hcl(h = seq(15, 375, length = 6), l = 65, c = 100)[1:5])
```
1-4 are 6/30, 5-8 are 7/14, 9-12 are 7/21, 13-16 are 7/27


# Fold change - specific metabolites
## calculating fold change and kmeans
```{r}
data_vis_fc <- data_vis %>% 
  select(-c("replicate")) %>% 
  filter(!str_detect(treatment, "Tote")) %>% 
  pivot_wider(names_from = treatment, values_from = nM, values_fn = mean) %>% 
  pivot_longer(cols = c(ZF, ZL, ZH, LL, LH, RL, RH)) %>% 
  mutate(fold_change = value/C)

data_vis_km <- data_vis %>% 
  pivot_wider(names_from = metabolite, values_from = nM) %>% 
  select(-c("treatment", "date")) %>% 
  column_to_rownames("replicate") %>% 
  data.matrix() %>%
  `[<-`(is.na(.), 0) %>%
  scale() %>%
  t() %>%
  kmeans(centers = 6) %>% 
  pluck("cluster") %>%
  as.data.frame() %>%
  set_names("cluster") %>%
  rownames_to_column("metabolite")

data_vis_cluster <- data_vis_fc %>% 
  left_join(data_vis_km) %>% 
  rename(control = C, treatment = name, nM = value) %>% 
  mutate(cluster = as.numeric(cluster))
```

## plotting
```{r}
data_vis_cluster %>% 
  na.omit() %>% 
  group_by(cluster, date, treatment) %>% 
  # filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  mutate(avg_fc = mean(fold_change)) %>% 
  ggplot(aes(x = date, y = log2(avg_fc), color = factor(treatment, levels = level_order))) + 
  geom_point() +
  geom_line() + 
  theme_bw() + 
  facet_wrap(~cluster) +
scale_color_manual(values = c("mediumpurple3","darkslategray2", "lightseagreen","gold", "lightcoral","orange", "brown3"),
                     labels = c("0N:1P, low dose Fe","0N:1P, low dose added", "0N:1P, high dose added",
                                "6N:1P, low dose added", "6N:1P, high dose added",
                                "16N:1P, low dose added", "16N:1P, high dose added"),
                     name = "Treatment") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) + 
  ylab("Average Fold Change") + 
  xlab("Date")

avg_fc_metab <- data_vis_cluster %>% 
  na.omit() %>% 
  group_by(cluster, date, treatment, metabolite) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  mutate(avg_fc = mean(fold_change)) %>% 
  ggplot(aes(x = date, y = log2(avg_fc), color = factor(treatment, levels = level_order), fill = metabolite)) + 
  geom_point() +
  geom_line() + 
  theme_bw() + 
  facet_wrap(~cluster) +
  scale_color_manual(values = c("mediumpurple3","darkslategray2", "lightseagreen","gold", "lightcoral","orange", "brown3"),
                     labels = c("0N:1P, low dose Fe","0N:1P, low dose added", "0N:1P, high dose added",
                                "6N:1P, low dose added", "6N:1P, high dose added",
                                "16N:1P, low dose added", "16N:1P, high dose added"),
                     name = "Treatment") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5), legend.position = "none") + 
  ylab("Average Fold Change") + 
  xlab("Date")

ggplotly(avg_fc_metab)
```



```{r}
nb.cols <- 35
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

plot_fc <- data_vis_cluster %>% 
  na.omit() %>% 
  # filter(!str_detect(treatment, "ZF|ZL|ZH")) %>% 
  group_by(metabolite, date, treatment) %>% 
  mutate(avg_fc = mean(fold_change)) %>% 
  filter(cluster == 6) %>% 
  ggplot() +
  geom_line(aes(x = date, y = log2(avg_fc), color = metabolite)) +
  scale_fill_manual(values = mycolors) + 
  facet_wrap(~treatment)

ggplotly(plot_fc)
```


# NMDS
```{r}
quant_nmds <- data_vis %>% 
  # arrange(desc(nM)) %>% 
  # group_by(metabolite, replicate) %>% 
  # slice(1) %>% 
  group_by(metabolite, treatment, date) %>%
  summarise(avg_nmol = mean(nM)) %>%
  ungroup() %>%
  # complete(metabolite, date, treatment) %>%
  # mutate(avg_nmol = replace_na(avg_nmol, 0)) %>%
  mutate(id = paste(date, treatment)) %>% 
  filter(!str_detect(id, "Tote|ZF|ZH|ZL"))

quant_mat <- quant_nmds %>%
  group_by(metabolite) %>%
  mutate(norm_conc = rank(avg_nmol)) %>%
  select(metabolite, norm_conc, id) %>% 
  pivot_wider(names_from = "metabolite", values_from = "norm_conc", values_fill = 0) %>%
  column_to_rownames("id") %>%
  data.matrix() 

mdsout <- quant_mat %>%
  metaMDS(k = 2, autotransform = FALSE)

mds_data <- metadataframe %>% 
  ungroup() %>% 
  mutate(timepoint = str_replace_all(timepoint, "July", "-7-22")) %>% 
  mutate(timepoint = str_replace_all(timepoint, "June", "-6-22")) %>% 
  mutate(date = as.Date(timepoint, format = "%d-%m-%y")) %>% 
  filter(str_detect(filename, "Smp")) %>% 
  mutate(treatment = str_remove(treatment, "\\d")) %>% 
  mutate(id = paste(date, treatment))

# %>% 
#   select(-c("replicate", "filename"))

mdsout$points %>%
  as.data.frame() %>%
  rownames_to_column("id") %>%
  left_join(mds_data) %>%
  mutate(date_fct = factor(date)) %>% 
  ggplot() +
  geom_point(aes(x=MDS1, y=MDS2, color=treatment, shape = date_fct), size=4) +
  scale_color_manual(values = c("plum2", "lightcoral", "gold", "brown3", "orange", "mediumpurple3",
                                "lightseagreen", "darkslategray2"),
                     labels = c("Control - no nutrients added", "6N:1P, high dose added", "6N:1P, low dose added",
                                "16N:1P, high dose added", "16N:1P, low dose added", "0N:1P:0Fe, high dose added",
                                "0N:1P, low dose added", "0N:1P, high dose added"),
                     name = "Treatment") +
  scale_shape_discrete(name = "Date") + 
  theme_bw() + 
  ggtitle("Variation of Metabolite Concentration (nM) Across Treatments")

mdsout$points %>%
  as.data.frame() %>%
  rownames_to_column("id") %>%
  left_join(mds_data) %>%
  mutate(date_fct = factor(date)) %>% 
  ggplot() +
  geom_point(aes(x=MDS1, y=MDS2, color=treatment, shape = date_fct), size=4) +
  scale_color_manual(values = c("plum2", "lightcoral", "gold", "brown3", "orange"),
                     labels = c("Control - no nutrients added", "6N:1P, high dose added", "6N:1P, low dose added",
                                "16N:1P, high dose added", "16N:1P, low dose added"),
                     name = "Treatment") +
  scale_shape_discrete(name = "Date") + 
  theme_bw() + 
  ggtitle("Variation of Metabolite Concentration (nM) Across Treatments")
```
Time is a very important factor, all the 6/30 dates are grouping together

```{r}
quant_nmds <- data_vis %>% 
  # arrange(desc(nM)) %>% 
  # group_by(metabolite, replicate) %>% 
  # slice(1) %>% 
  mutate(rate = str_extract(treatment, "R|L|Z|C|Tote")) %>% 
  group_by(rate, metabolite, date) %>%
  summarise(avg_nmol = mean(nM)) %>%
  ungroup() %>%
  # complete(metabolite, date, treatment) %>%
  # mutate(avg_nmol = replace_na(avg_nmol, 0)) %>%
  mutate(id = paste(date, rate)) %>% 
  filter(!str_detect(rate, "Tote"))

quant_mat <- quant_nmds %>%
  group_by(metabolite) %>%
  mutate(norm_conc = rank(avg_nmol)) %>%
  select(metabolite, norm_conc, id) %>% 
  pivot_wider(names_from = "metabolite", values_from = "norm_conc", values_fill = 0) %>%
  column_to_rownames("id") %>%
  data.matrix() 

mdsout <- quant_mat %>%
  metaMDS(k = 2, autotransform = FALSE)

mds_data <- metadataframe %>% 
  ungroup() %>% 
  mutate(timepoint = str_replace_all(timepoint, "July", "-7-22")) %>% 
  mutate(timepoint = str_replace_all(timepoint, "June", "-6-22")) %>% 
  mutate(date = as.Date(timepoint, format = "%d-%m-%y")) %>% 
  mutate(rate = str_extract(treatment, "R|L|Z|C")) %>% 
  filter(str_detect(filename, "Smp")) %>% 
  mutate(treatment = str_remove(treatment, "\\d")) %>% 
  mutate(id = paste(date, rate))

# %>% 
#   select(-c("replicate", "filename"))

mdsout$points %>%
  as.data.frame() %>%
  rownames_to_column("id") %>%
  left_join(mds_data) %>%
  mutate(date_fct = factor(date)) %>% 
  ggplot() +
  geom_point(aes(x=MDS1, y=MDS2, color=rate, shape = date_fct), size=4) +
  # scale_color_manual(values = c("plum2", "lightcoral", "gold", "brown3", "orange", "mediumpurple3",
  #                               "lightseagreen", "darkslategray2"),
  #                    labels = c("Control - no nutrients added", "6N:1P, high dose added", "6N:1P, low dose added",
  #                               "16N:1P, high dose added", "16N:1P, low dose added", "0N:0P, low dose of Fe",
  #                               "0N:1P, low dose added", "0N:1P, high dose added"),
  #                    name = "Treatment") +
  scale_shape_discrete(name = "Date") + 
  theme_bw() + 
  ggtitle("Variation of Metabolite Concentration (nM) Across Treatments")
```


```{r}
quant_nmds <- data_vis %>% 
  # arrange(desc(nM)) %>% 
  # group_by(metabolite, replicate) %>% 
  # slice(1) %>% 
  mutate(ratio = str_extract(treatment, "L|H|F|C|Tote\\d")) %>% 
  group_by(ratio, metabolite, date) %>%
  summarise(avg_nmol = mean(nM)) %>%
  ungroup() %>%
  # complete(metabolite, date, treatment) %>%
  # mutate(avg_nmol = replace_na(avg_nmol, 0)) %>%
  mutate(id = paste(date, ratio)) %>% 
  filter(!str_detect(ratio, "Tote"))

quant_mat <- quant_nmds %>%
  group_by(metabolite) %>%
  mutate(norm_conc = rank(avg_nmol)) %>%
  select(metabolite, norm_conc, id) %>% 
  pivot_wider(names_from = "metabolite", values_from = "norm_conc", values_fill = 0) %>%
  column_to_rownames("id") %>%
  data.matrix() 

mdsout <- quant_mat %>%
  metaMDS(k = 2, autotransform = FALSE)

mds_data <- metadataframe %>% 
  ungroup() %>% 
  mutate(timepoint = str_replace_all(timepoint, "July", "-7-22")) %>% 
  mutate(timepoint = str_replace_all(timepoint, "June", "-6-22")) %>% 
  mutate(date = as.Date(timepoint, format = "%d-%m-%y")) %>% 
  mutate(ratio = str_extract(treatment, "L|H|F|C\\d")) %>% 
  filter(str_detect(filename, "Smp")) %>% 
  mutate(treatment = str_remove(treatment, "\\d")) %>% 
  mutate(id = paste(date, ratio))

# %>% 
#   select(-c("replicate", "filename"))

mdsout$points %>%
  as.data.frame() %>%
  rownames_to_column("id") %>%
  left_join(mds_data) %>%
  mutate(date_fct = factor(date)) %>% 
  ggplot() +
  geom_point(aes(x=MDS1, y=MDS2, color=ratio, shape = date_fct), size=4) +
  # scale_color_manual(values = c("plum2", "lightcoral", "gold", "brown3", "orange", "mediumpurple3",
  #                               "lightseagreen", "darkslategray2"),
  #                    labels = c("Control - no nutrients added", "6N:1P, high dose added", "6N:1P, low dose added",
  #                               "16N:1P, high dose added", "16N:1P, low dose added", "0N:0P, low dose of Fe",
  #                               "0N:1P, low dose added", "0N:1P, high dose added"),
  #                    name = "Treatment") +
  scale_shape_discrete(name = "Date") + 
  theme_bw() + 
  ggtitle("Variation of Metabolite Concentration (nM) Across Treatments")
```


```{r}
data_vis %>% 
  group_by(metabolite, treatment, date) %>% 
  mutate(nmol_avg = mean(nM)) %>% 
  mutate(sd = sd(nM)) %>% 
  # slice(1) %>%
  distinct(nmol_avg, .keep_all = TRUE) %>%
  filter(str_detect(metabolite, "Guanine")) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  ggplot(aes(x = date, y = nmol_avg, color = factor(treatment, levels = level_order))) +
  geom_line() +
  geom_point() +
  geom_errorbar(aes(ymin = nmol_avg - sd, ymax = nmol_avg + sd)) + 
  scale_color_manual(values = c("gold", "lightcoral", "orange", "brown3"), 
                     labels = c("Lower (than Redfield) ratio of N:P, low dose added", "Lower (than Redfield) ratio of N:P, high dose added", 
                                "Redfield Ratio of N:P, low dose added", "Redfield Ratio of N:P, high dose added"), 
                     name = "Treatment") + 
  guides(fill=guide_legend(ncol=2)) + 
  theme_bw() + 
  ggtitle("Timeseries of Guanine, Summarised by Triplicate") + 
  xlab("Date") + 
  ylab("nM (nmol per L)") 
```

```{r}
data_vis %>% 
  group_by(metabolite, treatment, date) %>% 
  filter(str_detect(metabolite, "Trimethylamine N-oxide")) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  ggplot(aes(x = factor(date), y = nM, color = factor(treatment, levels = level_order))) +
  geom_boxplot() + 
  scale_color_manual(values = c("gold", "lightcoral", "orange", "brown3"), 
                     labels = c("Lower (than Redfield) ratio of N:P, low dose added", "Lower (than Redfield) ratio of N:P, high dose added", 
                                "Redfield Ratio of N:P, low dose added", "Redfield Ratio of N:P, high dose added"), 
                     name = "Treatment") + 
  guides(fill=guide_legend(ncol=2)) + 
  theme_bw() + 
  ggtitle("Timeseries of Trimethylamine N-Oxide") + 
  xlab("Date") + 
  ylab("nM (nmol per L)") 
```


```{r}
aminos <- read.csv(
  "https://raw.githubusercontent.com/IngallsLabUW/Ingalls_Standards/master/Ingalls_Lab_Standards.csv",
                              stringsAsFactors = FALSE, header = TRUE) %>% 
  filter(Compound_Type == "Amino Acid") %>% 
  select(metabolite = Compound_Name, emp_form = Empirical_Formula, metabolite_name = Compound_Name_Figure)

peridice_aas <- data_vis %>% 
  filter(metabolite %in% aminos$metabolite)

peridice_aas %>% 
  group_by(metabolite, treatment, date) %>% 
  filter(str_detect(treatment, "C|RH|RL|LH|LL")) %>% 
  ggplot(aes(x = factor(date), y = nM, color = factor(treatment, levels = level_order))) +
  geom_boxplot() + 
  scale_color_manual(values = c("plum2", "gold", "lightcoral", "orange", "brown3"),
                     labels = c("Control", "6N:1P, low dose added",
                                "6N:1P, high dose added",
                                "16N:1P, low dose added", "16N:1P, high dose added"),
                     name = "Treatment") +
  guides(fill=guide_legend(ncol=2)) + 
  theme_bw() + 
  xlab("Date") + 
  ylab("nM (nmol per L)") + 
  facet_wrap(~metabolite, scales = "free_y") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))
```
Collapse of ecosystem - amino acids track with c:n ratio


```{r}
nb.cols <- 5
mycolors <- colorRampPalette(brewer.pal(8, "Set3"))(nb.cols)

data_vis_aas <- data_vis_cluster %>% 
  filter(metabolite %in% aminos$metabolite)

# all aminos are in 5 except isoleucine and leucine

peridice_aas %>% 
  mutate(triplicate = str_extract(replicate, "\\d$")) %>% 
  # mutate(nmol_rank = rank(nM)) %>%
  mutate(nmol_rank=(nM-min(nM))/(max(nM)-min(nM))) %>%
  group_by(triplicate, treatment, date) %>% 
  mutate(avg_nmol = mean(nmol_rank)) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL|C")) %>% 
  filter(triplicate == 2) %>% 
  ggplot(aes(x = date, y = avg_nmol, color = treatment, group = interaction(treatment, triplicate))) + 
  scale_color_manual(values = c("gray50", "gold", "lightpink2", "darkorange", "brown3")) +
  geom_line() + 
  geom_point() + 
  ylab("Normalized Average Amino Acid Concentration (nM)") + 
  xlab("Date") +
  theme_bw() + 
  facet_wrap(~treatment) + 
  theme()

# color scheme - red and yellow, 
```

```{r}
library(readxl)
pcpn <- read_xlsx("metadata/peridice_pcpn.xlsx") %>% 
  select(tank = Tank, treatment = Treatment2, date = Date, pn = `PN (uM)`, pc = `PC (uM)`, cn = Cnratio) %>% 
  mutate(triplicate = str_extract(tank, "\\d")) %>% 
  filter(!str_detect(treatment, "Tote"))

pcpn %>% 
  mutate(cn_rank=(cn-min(cn))/(max(cn)-min(cn))) %>%
  filter(!str_detect(treatment, "Tote")) %>% 
  group_by(triplicate, treatment, date) %>%  
  # mutate(avg_cn = mean(cn)) %>% 
  # filter(triplicate == 2) %>% 
  ggplot(aes(x = date, y = cn, group = interaction(treatment, triplicate), color = treatment)) +
  geom_line() + 
  geom_point() + 
  # scale_color_manual(values = c("plum2", "gold", "lightcoral", "darkorange", "brown3"), name = "Treatment") +
  facet_wrap(~treatment) + 
  theme_bw() +  
  ylab("C to N Ratio") + 
  xlab("Date")
``` 

```{r}
osmos <- read.csv(
  "https://raw.githubusercontent.com/IngallsLabUW/Ingalls_Standards/master/Ingalls_Lab_Standards.csv",
                              stringsAsFactors = FALSE, header = TRUE) %>% 
  filter(Compound_Type == "Osmolyte") %>% 
  select(metabolite = Compound_Name, emp_form = Empirical_Formula, metabolite_name = Compound_Name_Figure)

peridice_osmos <- data_vis %>% 
  filter(metabolite %in% osmos$metabolite) %>% 
  mutate(metab_type = "Osmolyte")

peridice_osmos %>% 
  group_by(metabolite, treatment, date) %>% 
  # filter(!str_detect(treatment, "Tote")) %>% 
  filter(str_detect(treatment, "C|RH|RL|LH|LL")) %>% 
  filter(str_detect(metabolite, "2-O-alpha-D-Glucosylglycerol|Ectoine|Ethanolamine|Trimethylamine N-oxide")) %>% 
  ggplot(aes(x = factor(date), y = nM, color = factor(treatment, levels = level_order))) +
  geom_boxplot() + 
  scale_color_manual(values = c("plum2", "gold", "lightcoral", "darkorange", "brown3"),
                     labels = c("Control", "6N:1P, low dose added",
                                "6N:1P, high dose added",
                                "16N:1P, low dose added", "16N:1P, high dose added"),
                     name = "Treatment") +
  guides(fill=guide_legend(ncol=2)) + 
  theme_bw() + 
  xlab("Date") + 
  ylab("nM (nmol per L)") + 
  facet_wrap(~metabolite, scales = "free_y") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))
```

```{r}
nucleos <- read.csv(
  "https://raw.githubusercontent.com/IngallsLabUW/Ingalls_Standards/master/Ingalls_Lab_Standards.csv",
                              stringsAsFactors = FALSE, header = TRUE) %>% 
  filter(Compound_Type == "Nucleic Acid") %>% 
  select(metabolite = Compound_Name, emp_form = Empirical_Formula, metabolite_name = Compound_Name_Figure)

peridice_nucleos <- data_vis %>% 
  filter(metabolite %in% nucleos$metabolite) %>% 
  mutate(metab_type = "Nucleic Acid")

peridice_nucleos %>% 
  group_by(metabolite, treatment, date) %>% 
  filter(str_detect(treatment, "C|RH|RL|LH|LL")) %>% 
  ggplot(aes(x = factor(date), y = nM, color = factor(treatment, levels = level_order))) +
  geom_boxplot() + 
  scale_color_manual(values = c("plum2", "gold", "lightcoral", "orange", "brown3"),
                     labels = c("Control", "6N:1P, low dose added",
                                "6N:1P, high dose added",
                                "16N:1P, low dose added", "16N:1P, high dose added"),
                     name = "Treatment") +
  guides(fill=guide_legend(ncol=2)) + 
  theme_bw() + 
  xlab("Date") + 
  ylab("nM (nmol per L)") + 
  facet_wrap(~metabolite, scales = "free_y") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))
```

```{r}
sulfur <- read.csv(
  "https://raw.githubusercontent.com/IngallsLabUW/Ingalls_Standards/master/Ingalls_Lab_Standards.csv",
                              stringsAsFactors = FALSE, header = TRUE) %>% 
  filter(Compound_Type == "Sulfur") %>% 
  select(metabolite = Compound_Name, emp_form = Empirical_Formula, metabolite_name = Compound_Name_Figure)

peridice_sulf <- data_vis %>% 
  filter(metabolite %in% sulfur$metabolite) %>% 
  filter(metabolite != "2-Hydroxy-4-(methylthio)butyric acid") %>% 
  filter(metabolite != "L-Cysteinesulfinic acid") %>% 
  filter(metabolite != "Sodium taurocholate") %>% 
  filter(metabolite != "Amino hydroxypropanesulfonate") %>% 
  filter(metabolite != "3-Aminopropanesulfonate") %>% 
  filter(metabolite != "3-Sulfopyruvic acid") %>% 
  filter(metabolite != "2-keto-4-methylthiobutyric acid") %>% 
  mutate(metab_type = "Sulfur-Containing")

peridice_sulf %>% 
  group_by(metabolite, treatment, date) %>% 
  # filter(!str_detect(treatment, "Tote")) %>% 
  filter(str_detect(treatment, "C|RH|RL|LH|LL")) %>% 
  ggplot(aes(x = factor(date), y = nM, color = factor(treatment, levels = level_order))) +
  geom_boxplot() + 
  scale_color_manual(values = c("plum2", "gold", "lightcoral", "orange", "brown3"),
                     labels = c("Control", "6N:1P, low dose added",
                                "6N:1P, high dose added",
                                "16N:1P, low dose added", "16N:1P, high dose added"),
                     name = "Treatment") +
  guides(fill=guide_legend(ncol=2)) + 
  theme_bw() + 
  xlab("Date") + 
  ylab("nM (nmol per L)") + 
  facet_wrap(~metabolite, scales = "free_y") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) + 
  ggtitle("Concentrations of Sulfur-Containing Compounds")
```

```{r}
ranked_aas <- quant_nmds_heat %>% 
  # group_by(metabolite) %>%
  # filter(str_detect(treatment, "RH|RL|C|Tote")) %>% 
  # filter(avg_nmol >= 0.01) %>% 
  complete(metabolite, id, fill = list(avg_nmol = 0)) %>% 
  group_by(metabolite) %>% 
  mutate(norm_conc = rank(avg_nmol)) %>%
  select(metabolite, norm_conc, id) %>% 
  filter(metabolite %in% aminos$metabolite) 

data_vis_aas <- data_vis_cluster %>% 
  filter(metabolite %in% aminos$metabolite)

ranked_aas %>% 
  ggplot() +
  geom_raster(aes(x = metabolite, y = factor(id, levels = heat_level_order_date), fill = norm_conc)) +
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) +
  scale_fill_gradient2(low = "steelblue", mid = "grey90",
                       high = "goldenrod1", midpoint = 16.5,
                       breaks=c(0, 16.5, 33), labels=c(
                         "Very little of a given compound",
                         "Median compound abundance",
                         "Compound maximum"
                       )) +
  new_scale_fill() + 
  # scale_fill_discrete(guide="none") +
  # theme_void() +
  theme(legend.position = "top",
        legend.key.width = unit(dev.size()[1] / 6, "inches"),
        legend.text = element_text(size=8))
```

Highest in 7/14 RH/RL. Only treatments where amino acid concentrations subsist are in RH and RL.

Temporal scale shows peak at strong temporal pattern. Highest concentrations of all amino acids at 7/14. Physiological change versus community composition change. Amino acids are part of the physiological change these microbes are experiencing with these changes in nutrient concentrations. L-Aspartic Acid? 

```{r}
data_vis %>% 
  group_by(metabolite, treatment, date) %>% 
  mutate(nmol_avg = mean(nM)) %>% 
  # mutate(sd = sd(nM)) %>% 
  # slice(1) %>%
  distinct(nmol_avg, .keep_all = TRUE) %>%
  filter(str_detect(metabolite, "4-Aminobutyric acid")) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  ggplot(aes(x = date, y = nmol_avg, color = factor(treatment, levels = level_order))) +
  geom_line() +
  geom_point() +
  # geom_errorbar(aes(ymin = nmol_avg - sd, ymax = nmol_avg + sd)) + 
  scale_color_manual(values = c("gold", "lightcoral", "orange", "brown3"), 
                     labels = c("Lower (than Redfield) ratio of N:P, low dose added", "Lower (than Redfield) ratio of N:P, high dose added", 
                                "Redfield Ratio of N:P, low dose added", "Redfield Ratio of N:P, high dose added"), 
                     name = "Treatment") + 
  guides(fill=guide_legend(ncol=2)) + 
  theme_bw() + 
  ggtitle("Timeseries of L-Isoleucine, Summarised by Triplicate") + 
  xlab("Date") + 
  ylab("nM (nmol per L)") 
```


```{r}
ranked_sulf <- quant_nmds_heat %>% 
  # group_by(metabolite) %>%
  # filter(str_detect(treatment, "RH|RL|C|Tote")) %>% 
  # filter(avg_nmol >= 0.01) %>% 
  complete(metabolite, id, fill = list(avg_nmol = 0)) %>% 
  group_by(metabolite) %>% 
  mutate(norm_conc = rank(avg_nmol)) %>%
  select(metabolite, norm_conc, id) %>% 
  filter(metabolite %in% sulfur$metabolite) 


ranked_sulf %>% 
  ggplot() +
  geom_raster(aes(x = metabolite, y = factor(id, levels = heat_level_order_date), fill = norm_conc)) +
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) +
  scale_fill_gradient2(low = "steelblue", mid = "grey90",
                       high = "goldenrod1", midpoint = 16.5,
                       breaks=c(0, 16.5, 33), labels=c(
                         "Very little of a given compound",
                         "Median compound abundance",
                         "Compound maximum"
                       )) +
  new_scale_fill() + 
  # scale_fill_discrete(guide="none") +
  # theme_void() +
  theme(legend.position = "top",
        legend.key.width = unit(dev.size()[1] / 6, "inches"),
        legend.text = element_text(size=8))
```

```{r}
ranked_nuc <- quant_nmds_heat %>% 
  # group_by(metabolite) %>%
  # filter(str_detect(treatment, "RH|RL|C|Tote")) %>% 
  # filter(avg_nmol >= 0.01) %>% 
  complete(metabolite, id, fill = list(avg_nmol = 0)) %>% 
  group_by(metabolite) %>% 
  mutate(norm_conc = rank(avg_nmol)) %>%
  select(metabolite, norm_conc, id) %>% 
  filter(metabolite %in% nucleos$metabolite) 


ranked_nuc %>% 
  filter(!str_detect(id, "Tote")) %>% 
  ggplot() +
  geom_raster(aes(x = metabolite, y = factor(id, levels = heat_level_order_date), fill = norm_conc)) +
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) +
  scale_fill_gradient2(low = "steelblue", mid = "grey90",
                       high = "goldenrod1", midpoint = 16,
                       breaks=c(0, 16, 32), labels=c(
                         "Very little of a given compound",
                         "Median compound abundance",
                         "Compound maximum"
                       )) +
  new_scale_fill() + 
  # scale_fill_discrete(guide="none") +
  # theme_void() +
  theme(legend.position = "top",
        legend.key.width = unit(dev.size()[1] / 6, "inches"),
        legend.text = element_text(size=8)) +
  theme_bw()
```

# Temporal versus Spatial Changes
clusters 5 and 6
```{r}
nb.cols <- 35
mycolors <- colorRampPalette(brewer.pal(8, "Set1"))(nb.cols)

plot_fc <- data_vis_cluster %>% 
  na.omit() %>% 
  filter(!str_detect(treatment, "ZF|ZL|ZH|Tote")) %>% 
  group_by(metabolite, date, treatment) %>% 
  mutate(avg_fc = mean(fold_change)) %>% 
  filter(cluster == 4) %>% 
  ggplot() +
  geom_line(aes(x = date, y = avg_fc, color = metabolite)) +
  scale_fill_manual(values = mycolors) + 
  facet_wrap(~treatment)

ggplotly(plot_fc)
```


```{r}
# citric acid and isocitric acid and chitobiose
data_vis %>% 
  group_by(metabolite, treatment, date) %>% 
  filter(str_detect(metabolite, "Citric acid")) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  mutate(nmol_avg = mean(nM)) %>% 
  mutate(sd = sd(nM)) %>% 
  distinct(nmol_avg, .keep_all = TRUE) %>%
  mutate(nmol_avg = replace_na(nmol_avg, 0)) %>% 
  ggplot(aes(x = date, y = nmol_avg, color = factor(treatment, levels = level_order))) +
  geom_line() +
  geom_point() +
  # geom_errorbar(aes(ymin = nmol_avg - sd, ymax = nmol_avg + sd)) + 
  scale_color_manual(values = c("gold", "lightcoral", "orange", "brown3"), 
                     labels = c("6N:1P, low dose added", "6N:1P, high dose added", 
                                "16N:1P, low dose added", "16N:1P, high dose added"), 
                     name = "Treatment") + 
  guides(fill=guide_legend(ncol=2)) + 
  theme_bw() + 
  ggtitle("Timeseries of Citric Acid, Summarised by Triplicate") + 
  xlab("Date") + 
  ylab("nM (nmol per L)") 


# chitobiose is associated with vibrio cholerae?
data_vis %>% 
  group_by(metabolite, treatment, date) %>% 
  filter(str_detect(metabolite, "Gonyol")) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  mutate(nmol_avg = mean(nM)) %>% 
  mutate(sd = sd(nM)) %>% 
  distinct(nmol_avg, .keep_all = TRUE) %>%
  mutate(nmol_avg = replace_na(nmol_avg, 0)) %>% 
  ggplot(aes(x = date, y = nmol_avg, color = factor(treatment, levels = level_order))) +
  geom_line() +
  geom_point() +
  # geom_errorbar(aes(ymin = nmol_avg - sd, ymax = nmol_avg + sd)) + 
  scale_color_manual(values = c("gold", "lightcoral", "orange", "brown3"), 
                     labels = c("6N:1P, low dose added", "6N:1P, high dose added", 
                                "16N:1P, low dose added", "16N:1P, high dose added"), 
                     name = "Treatment") + 
  guides(fill=guide_legend(ncol=2)) + 
  theme_bw() + 
  ggtitle("Timeseries of Chitobiose, Summarised by Triplicate") + 
  xlab("Date") + 
  ylab("nM (nmol per L)") 

```

```{r}
data_vis %>% 
  group_by(metabolite, treatment, date) %>% 
  # mutate(nmol_avg = median(nM)) %>% 
  filter(str_detect(metabolite, "Deoxyadenosine|Deoxyguanosine")) %>% 
  filter(!str_detect(treatment, "Tote|C|ZH|ZL|ZF")) %>% 
  ggplot(aes(x = date, y = nM, color = factor(treatment, levels = level_order), group = interaction(date, treatment))) +
  geom_boxplot() + 
  scale_color_manual(values = c("gold", "lightcoral","orange", "brown3"),
                     labels = c("6N:1P, low dose added", "6N:1P, high dose added",
                                "16N:1P, low dose added", "16N:1P, high dose added"),
                     name = "Treatment") +
  guides(fill=guide_legend(ncol=2)) + 
  scale_y_log10() + 
  facet_wrap(~metabolite, ncol = 1, scales = "free_y") +
  theme_bw() + 
  xlab("Date") + 
  ylab("nM (nmol per L)") 

data_vis %>% 
  group_by(metabolite, treatment, date) %>% 
  # mutate(nmol_avg = median(nM)) %>% 
  filter(str_detect(metabolite, "Gonyol|Dimethylsulfonioacetate|Dimethylsulfoniopropionate")) %>% 
  filter(!str_detect(treatment, "Tote|C|ZH|ZL|ZF")) %>% 
  ggplot(aes(x = date, y = nM, color = factor(treatment, levels = level_order), group = interaction(date, treatment))) +
  geom_boxplot() + 
  scale_color_manual(values = c("gold", "lightcoral","orange", "brown3"),
                     labels = c("6N:1P, low dose added", "6N:1P, high dose added",
                                "16N:1P, low dose added", "16N:1P, high dose added"),
                     name = "Treatment") +
  guides(fill=guide_legend(ncol=2)) + 
  scale_y_log10() + 
  facet_wrap(~metabolite, ncol = 1, scales = "free_y") +
  theme_bw() + 
  xlab("Date") + 
  ylab("nM (nmol per L)") 
```


```{r}
peridice_metab_groups <- peridice_aas %>% 
  mutate(metab_type = "Amino Acid") %>% 
  rbind(peridice_nucleos, peridice_sulf, peridice_osmos) 

peridice_metab_groups %>% 
  mutate(nmol_rank=(nM-min(nM))/(max(nM)-min(nM))) %>%
  group_by(metab_type, treatment, date) %>% 
  mutate(avg_nmol = mean(nmol_rank)) %>% 
  distinct(avg_nmol, .keep_all = TRUE) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL|C")) %>% 
  ggplot() + 
  geom_col(aes(x = date, y = avg_nmol, fill = metab_type), position = "stack") + 
  scale_fill_manual(values = c("palegreen4", "palegreen3", "goldenrod1", "darkgoldenrod3"), name = "Metabolite Type") + 
  facet_grid(~metab_type~factor(treatment, levels = level_order), scales = "free_y") + 
  theme_bw() + 
  xlab("Treatment Group")
``` 
What is the significance of RL having more sulfur-containing metabs? RH has more osmolytes, nucleic acids enriched at the end of the experiment? 

```{r}
nb.cols <- 11
mycolors <- colorRampPalette(brewer.pal(8, "Paired"))(nb.cols)

metab_groups <- read.csv(
  "https://raw.githubusercontent.com/IngallsLabUW/Ingalls_Standards/master/Ingalls_Lab_Standards.csv",
                              stringsAsFactors = FALSE, header = TRUE) %>% 
  select(metabolite = Compound_Name, emp_form = Empirical_Formula, metab_type = Compound_Type)

data_vis_groups <- data_vis %>% 
  left_join(metab_groups, by = "metabolite") %>% 
  # mutate(metab_type = str_replace_all(metab_type, " - degraded", "")) %>% 
  # mutate(metab_type = str_replace_all(metab_type, " derivative", "")) %>% 
  # mutate(metab_type = str_replace_all(metab_type, " synthesis", "")) %>% 
  mutate(metab_type = str_replace_all(metab_type, " - collagen", "")) %>% 
  filter(!str_detect(metab_type, "Cursed|Amino sugar|Alkaloid|Energy charge|Amine|Environmental contaminant|Organic Acid|Arsenic|Tripeptide|Peptidoglycan|Lignin|Vitamin|Antioxidant|Dipeptide|Calvin|Nucleotide|Intra|Redox|Quorum")) %>% 
  mutate(metab_type = str_replace_all(metab_type, " - nitrogenous", "")) %>% 
  mutate(metab_type = str_replace_all(metab_type, " alchohol", "")) %>% 
  mutate(metab_type = str_replace_all(metab_type, " phosphate", "")) %>% 
  mutate(metab_type = str_replace_all(metab_type, "Glycine metabolite", "Amino Acid")) %>% 
  mutate(metab_type = str_replace_all(metab_type, "Betaine", "Amino Acid")) %>% 
  mutate(metab_type = str_replace_all(metab_type, "Purine|Pyrimidine", "Nucleoside"))
  
data_vis_groups %>% 
  group_by(metabolite, treatment, date) %>% 
  mutate(nmol_avg = mean(nM)) %>% 
  distinct(nmol_avg, .keep_all = TRUE) %>% 
  filter(str_detect(treatment, "LL|LH|RL|RH")) %>% 
  # filter(!str_detect(treatment, "Tote")) %>% 
  # filter(str_detect(replicate, "14July|27July")) %>% 
  ggplot() +
  geom_col(aes(x = factor(date), y = nmol_avg, fill = metab_type), position = "fill") +
  scale_fill_manual(values = mycolors, name = "Metabolite Type") + 
  facet_grid(~treatment, scales = "free_y") + 
  theme_bw() + 
  ylab("Metabolite Composition (nM)") + 
  xlab("Date")

```

where i want to take peridice scope video - story about time, story about metabolites skyrocketing up at 2 weeks, the ones that persist, what they are. 

Time - defines physiological changes versus those of community composition. 
2 col plots, metabolomes at 2 weeks and 4 weeks?

```{r}
diatom_metab <- data.frame(metabolite = c("Chitobiose", "Homarine", "Sucrose", "Hydroxyproline",
                                          "(R)-2,3-Dihydroxypropane-1-sulfonate", "Dimethylsulfonioacetate",
                                          "O-Propionylcarnitine", "O-Acetylcarnitine", "Thymine", "cis-Aconitic acid", 
                                          "Adenine", "L-Arginosuccinic acid")) %>% 
  mutate(metab_type = "Eukaryotes")

dino_metab <- data.frame(metabolite = c("Gonyol", "Dimethylsulfoniopropionate", "Proline betaine")) %>% 
  mutate(metab_type = "Dinoflagellate")

metab_taxon <- diatom_metab %>% 
  rbind(dino_metab)

data_vis_taxon <- data_vis %>% 
  left_join(metab_taxon)
  


data_vis_taxon %>%  
  na.omit() %>% 
  group_by(treatment, date, metab_type) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  # distinct(avg_nmol, .keep_all = TRUE) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL|C|Tote")) %>% 
  mutate(treatment = str_replace(treatment, "Tote", "T0")) %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "fill") + 
  scale_fill_manual(values = mycolors, name = "Taxonomic Group") + 
  facet_grid(~factor(treatment, levels = level_order), scales = "free", space = "free") + 
  theme_bw() + 
  ylab("Normalized Average Metabolite Concentration (nM)") + 
  xlab("Date") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))

```
# what is a better way to do this?
pull in xlsx file, export?

```{r}
cyano_euks <- data.frame(cyano_euks = c("Chitobiose", "Homarine", "Hydroxyectoine", "Sucrose", "Hydroxylproline")) %>% 
  mutate(metab_type = "cyano_euks")

only_euks <- data.frame(only_euks = c("O-Acetylcarnitine", "(R)-2,3-Dihydroxypropane-1-sulfonate",
                                 "Dimethylsulfonioacetate", "Gonyol", "Isethionic acid",	
                                 "N-Acetyltaurine",	"N-Methyltaurine", "O-Propionylcarnitine")) %>% 
  mutate(metab_type = "only euks")

bac_arch <- data.frame(bac_arch = c("cAMP",	"Cyclic.diGMP",	"Methylphosphonic.Acid",	"NADH",	"Propionyl.CoA", "Thymine-HILIC",	"X2iP-RP"))
```

```{r}
nb.cols <- 7
mycolors <- colorRampPalette(brewer.pal(8, "Paired"))(nb.cols)

data_vis_rat <- data_vis_taxon %>% 
  mutate(rate = str_extract(treatment, "H$|F$|C$|L$|Tote$")) %>% 
  mutate(ratio = str_extract(treatment, "R|L|Z|C|Tote")) %>% 
  mutate(rate = str_replace(rate, "Tote", "T0")) %>% 
  mutate(ratio = str_replace(ratio, "Tote", "T0"))

data_vis_rat %>% 
  mutate(ratio = str_replace(ratio, "L", "6N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "R", "16N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "Z", "0N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "C", "Control")) %>% 
  group_by(date, ratio, metab_type) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  distinct(avg_nmol, .keep_all = TRUE) %>% 
  na.omit() %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "stack") + 
  scale_fill_manual(values = mycolors, name = "Taxonomic Group") + 
  facet_grid(~factor(ratio, levels = c("T0", "Control", "0N:1P","6N:1P", "16N:1P")), scales = "free", space = "free") + 
  ylab("Average Metabolite Concentration (nM)") + 
  xlab("Date") + 
  theme_bw() + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))

data_vis_rat %>% 
  mutate(rate = str_replace(rate, "H", "High Dose")) %>% 
  mutate(rate = str_replace(rate, "L", "Low Dose")) %>% 
  mutate(rate = str_replace(rate, "C", "Control")) %>% 
  filter(!str_detect(rate, "F")) %>% 
  group_by(date, rate, metab_type) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  distinct(avg_nmol, .keep_all = TRUE) %>% 
  na.omit() %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "stack") + 
  scale_fill_manual(values = mycolors, name = "Taxonomic Group") + 
  facet_grid(~factor(rate, levels = c("T0", "Control", "Low Dose", "High Dose")), scales = "free", space = "free") + 
  ylab("Average Metabolite Concentration (nM)") + 
  xlab("Date") + 
  theme_bw() + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))


data_vis_rat %>% 
  group_by(date, treatment, metab_type) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  mutate(ratio = str_replace(ratio, "L", "6N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "R", "16N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "Z", "0N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "C", "Control")) %>% 
  mutate(rate = str_replace(rate, "H", "High Dose")) %>% 
  mutate(rate = str_replace(rate, "L", "Low Dose")) %>% 
  mutate(rate = str_replace(rate, "C", "Control")) %>% 
  mutate(rate = str_replace(rate, "F", "High Dose")) %>% 
  filter(!str_detect(treatment, "Tote|C")) %>% 
  distinct(avg_nmol, .keep_all = TRUE) %>% 
  na.omit() %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "fill") + 
  scale_fill_manual(values = mycolors, name = "Metabolite Function") +
  facet_grid(~rate~ratio) + 
  theme_bw() + 
  ggtitle("Ratio and Rate of Nutrient Addition Impacts on Microbial Community")+
  ylab("Average Metabolite Concentrations (nM)") + 
  xlab("Date") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))
```

```{r}
nb.cols <- 7
mycolors <- colorRampPalette(brewer.pal(8, "Paired"))(nb.cols)

data_vis_rat_group <- data_vis_groups %>% 
  mutate(rate = str_extract(treatment, "H$|F$|C$|L$|Tote$")) %>% 
  mutate(ratio = str_extract(treatment, "R|L|Z|C|Tote")) %>% 
  mutate(rate = str_replace(rate, "Tote", "T0")) %>% 
  mutate(ratio = str_replace(ratio, "Tote", "T0"))
 
data_vis_rat_group %>% 
  group_by(date, ratio, metab_type) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  mutate(ratio = str_replace(ratio, "L", "6N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "R", "16N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "Z", "0N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "C", "Control")) %>% 
  distinct(avg_nmol, .keep_all = TRUE) %>% 
  na.omit() %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "stack") + 
  scale_fill_manual(values = mycolors, name = "Metabolite Function") +
  facet_grid(~factor(ratio, levels = c("T0","Control", "0N:1P", "6N:1P", "16N:1P")), scales = "free", space = "free") + 
  theme_bw() + 
  ggtitle("Ratio of Nutrients Impacts on Metabolite Composition")+
  ylab("Average Metabolite Concentrations (nM)") + 
  xlab("Date") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))

data_vis_rat_group %>% 
  filter(!str_detect(treatment, "F")) %>% 
  mutate(rate = str_replace(rate, "H", "High Dose")) %>% 
  mutate(rate = str_replace(rate, "L", "Low Dose")) %>% 
  mutate(rate = str_replace(rate, "C", "Control")) %>% 
  group_by(date, treatment, metab_type) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  distinct(avg_nmol, .keep_all = TRUE) %>% 
  na.omit() %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "stack") + 
  scale_fill_manual(values = mycolors, name = "Metabolite Function") +
  facet_grid(~factor(rate, levels = c("T0","Control", "Low Dose", "High Dose")), scales = "free", space = "free") + 
  theme_bw() + 
  ggtitle("Rate of Nutrients Impacts on Metabolite Composition")+
  ylab("Average Metabolite Concentrations (nM)") + 
  xlab("Date") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))
```

```{r}
nb.cols <- 7
mycolors <- colorRampPalette(brewer.pal(8, "Paired"))(nb.cols)

data_vis_rat_taxon <- data_vis_groups %>% 
  mutate(rate = str_extract(treatment, "H$|F$|C$|L$|Tote$")) %>% 
  mutate(ratio = str_extract(treatment, "R|L|Z|C")) 

data_vis_rat_taxon %>% 
  group_by(date, treatment, metab_type) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  mutate(ratio = str_replace(ratio, "L", "6N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "R", "16N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "Z", "0N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "C", "Control")) %>% 
  distinct(avg_nmol, .keep_all = TRUE) %>% 
  na.omit() %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "stack") + 
  scale_fill_manual(values = mycolors, name = "Metabolite Function") +
  facet_grid(~ratio, scales = "free") + 
  theme_bw() + 
  ggtitle("Ratio of Nutrients Impacts on Metabolite Composition")+
  ylab("Average Metabolite Concentrations (nM)") + 
  xlab("Date") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))

data_vis_rat_group %>% 
  filter(!str_detect(treatment, "F")) %>% 
  mutate(rate = str_replace(rate, "H", "High Dose")) %>% 
  mutate(rate = str_replace(rate, "L", "Low Dose")) %>% 
  mutate(rate = str_replace(rate, "C", "Control")) %>% 
  group_by(date, rate, metab_type) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  distinct(avg_nmol, .keep_all = TRUE) %>% 
  na.omit() %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "fill") + 
  scale_fill_manual(values = mycolors, name = "Metabolite Function") +
  facet_grid(~rate) + 
  theme_bw() + 
  ggtitle("Rate of Nutrients Impacts on Metabolite Composition")+
  ylab("Average Metabolite Concentrations (nM)") + 
  xlab("Date") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))
```

```{r}
data_vis_rat_taxon %>% 
  group_by(date, treatment, metab_type) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  mutate(ratio = str_replace(ratio, "L", "6N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "R", "16N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "Z", "0N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "C", "Control")) %>% 
  mutate(rate = str_replace(rate, "H", "High Dose")) %>% 
  mutate(rate = str_replace(rate, "L", "Low Dose")) %>% 
  mutate(rate = str_replace(rate, "C", "Control")) %>% 
  mutate(rate = str_replace(rate, "F", "High Dose")) %>% 
  filter(!str_detect(treatment, "C")) %>% 
  distinct(avg_nmol, .keep_all = TRUE) %>% 
  na.omit() %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "fill") + 
  scale_fill_manual(values = mycolors, name = "Metabolite Function") +
  facet_grid(~rate~ratio) + 
  theme_bw() + 
  ggtitle("Ratio of Nutrients Impacts on Metabolite Composition")+
  ylab("Average Metabolite Concentrations (nM)") + 
  xlab("Date") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))

```
```{r}
data_vis_rat_taxon %>% 
  group_by(date, treatment, metab_type) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  mutate(ratio = str_replace(ratio, "L", "6N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "R", "16N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "Z", "0N:1P")) %>% 
  mutate(ratio = str_replace(ratio, "C", "Control")) %>% 
  mutate(rate = str_replace(rate, "H", "High Dose")) %>% 
  mutate(rate = str_replace(rate, "L", "Low Dose")) %>% 
  mutate(rate = str_replace(rate, "C", "Control")) %>% 
  mutate(rate = str_replace(rate, "F", "High Dose")) %>% 
  filter(str_detect(treatment, "C")) %>% 
  distinct(avg_nmol, .keep_all = TRUE) %>% 
  na.omit() %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "stack") + 
  scale_fill_manual(values = mycolors, name = "Metabolite Function") +
  facet_grid(~rate~ratio) + 
  ylim(0,3.5) + 
  theme_bw() + 
  # ggtitle("Ratio of Nutrients Impacts on Metabolite Composition")+
  ylab("Average Metabolite Concentrations (nM)") + 
  xlab("Date") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))

```

```{r}
data_vis_cluster_sheet <- data_vis_cluster %>% 
  left_join(metab_groups) %>%
  mutate(metab_type = str_replace_all(metab_type, " - degraded", "")) %>% 
  mutate(metab_type = str_replace_all(metab_type, " derivative", "")) %>% 
  mutate(metab_type = str_replace_all(metab_type, " synthesis", "")) %>% 
  mutate(metab_type = str_replace_all(metab_type, " - collagen", "")) %>% 
  filter(!str_detect(metab_type, "Cursed|Amino sugar|Alkaloid|Energy charge|Amine|Environmental contaminant|Organic Acid|Arsenic|Tripeptide|Peptidoglycan|Lignin|Vitamin|Antioxidant|Dipeptide|Calvin|Nucleotide|Intra|Redox|Quorum")) %>% 
  mutate(metab_type = str_replace_all(metab_type, " - nitrogenous", "")) %>% 
  mutate(metab_type = str_replace_all(metab_type, " alchohol", "")) %>% 
  mutate(metab_type = str_replace_all(metab_type, " phosphate", "")) %>% 
  mutate(metab_type = str_replace_all(metab_type, "Glycine metabolite", "Amino Acid")) %>% 
  mutate(metab_type = str_replace_all(metab_type, "Betaine", "Amino Acid")) %>% 
  mutate(metab_type = str_replace_all(metab_type, "Purine|Pyrimidine", "Nucleoside"))

data_vis_cluster_sheet %>% 
  na.omit() %>% 
  filter(!str_detect(treatment, "ZF|ZL|ZH|Tote")) %>% 
  group_by(cluster, metab_type, date, treatment) %>% 
  mutate(avg_fc = mean(fold_change)) %>% 
  distinct(avg_fc) %>% 
  # filter(cluster == 4) %>% 
  ggplot() +
  geom_line(aes(x = date, y = avg_fc, color = metab_type)) +
  scale_fill_manual(values = mycolors) + 
  facet_wrap(~cluster, scales = "free")

data_vis_cluster_sheet %>% 
  na.omit() %>% 
  group_by(cluster, metab_type, date, treatment) %>% 
  # filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  mutate(avg_fc = mean(fold_change)) %>% 
  ggplot(aes(x = date, y = avg_fc, color = factor(treatment, levels = level_order))) + 
  geom_point() +
  geom_line() + 
  theme_bw() + 
  facet_wrap(~cluster) +
scale_color_manual(values = c("mediumpurple3","darkslategray2", "lightseagreen","gold", "lightcoral","orange", "brown3"),
                     labels = c("0N:1P:0Fe, high dose added","0N:1P, low dose added", "0N:1P, high dose added",
                                "6N:1P, low dose added", "6N:1P, high dose added",
                                "16N:1P, low dose added", "16N:1P, high dose added"),
                     name = "Treatment") +
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) + 
  ylab("Average Fold Change") + 
  xlab("Date")

data_vis_cluster_sheet %>% 
  group_by(metab_type, cluster, treatment, date) %>% 
  mutate(avg_nmol = mean(nM)) %>% 
  filter(cluster == 1) %>% 
  ggplot() + 
  geom_col(aes(x = factor(date), y = avg_nmol, fill = metab_type), position = "stack") + 
  scale_fill_manual(values = mycolors, name = "Metabolite Function") +
  facet_grid(~treatment) + 
  ylim(0,3.5) + 
  theme_bw() + 
  # ggtitle("Ratio of Nutrients Impacts on Metabolite Composition")+
  ylab("Average Metabolite Concentrations (nM)") + 
  xlab("Date") + 
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5))

data_vis_cluster_sheet %>% 
  filter(cluster == 6) %>% 
  distinct(metabolite, .keep_all = TRUE)
```


# Chemistry :)
```{r}
data_vis_emp_forms <- read.csv(
  "https://raw.githubusercontent.com/IngallsLabUW/Ingalls_Standards/master/Ingalls_Lab_Standards.csv",
                              stringsAsFactors = FALSE, header = TRUE) %>% 
  select(metabolite = Compound_Name, emp_form = Empirical_Formula, metab_type = Compound_Type)

data_vis_chem <- data_vis %>% 
  left_join(data_vis_emp_forms, by = "metabolite")

# kegg code to pull in shape or other info?
```

```{r}
metab_type_fc_plot <- data_vis_cluster_sheet %>% 
  na.omit() %>% 
  group_by(cluster, date, treatment, metab_type) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  mutate(avg_fc = mean(fold_change)) %>% 
  distinct(avg_fc, .keep_all = TRUE) %>% 
  # filter(avg_fc >= 1) %>% 
  ggplot(aes(x = date, y = log2(avg_fc), color = metab_type, shape = treatment)) + 
  geom_point() + 
  geom_line() + 
  theme_bw() + 
  facet_wrap(~cluster) +
# scale_color_manual(values = c("mediumpurple3","darkslategray2", "lightseagreen","gold", "lightcoral","orange", "brown3"),
#                      labels = c("0N:1P:0Fe, high dose added","0N:1P, low dose added", "0N:1P, high dose added",
#                                 "6N:1P, low dose added", "6N:1P, high dose added",
#                                 "16N:1P, low dose added", "16N:1P, high dose added"),
#                      name = "Treatment") +
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) + 
  ylab("Average Fold Change") + 
  xlab("Date")

ggplotly(metab_type_fc_plot)

fig <- data_vis_cluster_sheet %>% 
  filter(metab_type == "Sulfur")

nb.cols <- 23
mycolors <- colorRampPalette(brewer.pal(8, "Paired"))(nb.cols)

interactive_sulf <- data_vis_cluster_sheet %>% 
  group_by(date, treatment, metabolite) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  mutate(avg_fc = mean(fold_change)) %>% 
  filter(metab_type == "Sulfur") %>% 
  ggplot(aes(x = date, y = log2(avg_fc), color = metabolite, shape = treatment)) + 
  geom_point() + 
  geom_line() + 
  theme_bw() + 
  facet_wrap(~cluster) +
  scale_color_manual(values = mycolors) +
# scale_color_manual(values = c("mediumpurple3","darkslategray2", "lightseagreen","gold", "lightcoral","orange", "brown3"),
#                      labels = c("0N:1P:0Fe, high dose added","0N:1P, low dose added", "0N:1P, high dose added",
#                                 "6N:1P, low dose added", "6N:1P, high dose added",
#                                 "16N:1P, low dose added", "16N:1P, high dose added"),
#                      name = "Treatment") +
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) + 
  ylab("Average Fold Change") + 
  xlab("Date")

ggplotly(interactive_sulf)

data_vis_cluster_sheet %>% 
  filter(str_detect(metabolite, "Choline"))

int_clust <- data_vis_cluster_sheet %>% 
  group_by(date, treatment, metabolite) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  mutate(avg_fc = mean(fold_change)) %>% 
  # filter(metab_type == "Sulfur") %>% 
  ggplot(aes(x = date, y = log2(avg_fc), color = metabolite, shape = treatment)) +
  geom_line() + 
  geom_point() + 
  theme_bw() + 
  facet_grid(~cluster~treatment) +
  # scale_color_manual(values = mycolors) +
# scale_color_manual(values = c("mediumpurple3","darkslategray2", "lightseagreen","gold", "lightcoral","orange", "brown3"),
#                      labels = c("0N:1P:0Fe, high dose added","0N:1P, low dose added", "0N:1P, high dose added",
#                                 "6N:1P, low dose added", "6N:1P, high dose added",
#                                 "16N:1P, low dose added", "16N:1P, high dose added"),
#                      name = "Treatment") +
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) + 
  ylab("Average Fold Change") + 
  xlab("Date")

ggplotly(int_clust)

```

normalize to POC - send to josh, use tfinalsx

```{r}
sulfur_fc <- data_vis_cluster_sheet %>% 
  na.omit() %>% 
  group_by(date, treatment, metabolite) %>% 
  filter(str_detect(metab_type, "Sulfur")) %>% 
  mutate(avg_fc = mean(fold_change)) %>% 
  distinct(avg_fc, .keep_all = TRUE) %>% 
  filter(str_detect(treatment, "RH|RL|LH|LL")) %>% 
  ggplot(aes(x = date, y = log2(avg_fc), color = metabolite)) + 
  geom_line() + 
  theme_bw() + 
  facet_wrap(~cluster~treatment) +
# scale_color_manual(values = c("mediumpurple3","darkslategray2", "lightseagreen","gold", "lightcoral","orange", "brown3"),
#                      labels = c("0N:1P:0Fe, high dose added","0N:1P, low dose added", "0N:1P, high dose added",
#                                 "6N:1P, low dose added", "6N:1P, high dose added",
#                                 "16N:1P, low dose added", "16N:1P, high dose added"),
#                      name = "Treatment") +
  theme(axis.text.x=element_text(angle = 90, vjust = 0.5)) + 
  ylab("Average Fold Change") + 
  xlab("Date")

ggplotly(sulfur_fc)
```

dms-ac, cluster 1, concentrated in Ls not Rs. 


```{r}
pcpn <- read_xlsx("metadata/peridice_pcpn.xlsx") %>% 
  select(tank = Tank, date = Date, pn_um = `PN (uM)`, pc_um = `PC (uM)`, cn_ratio = Cnratio)
  # filter(str_detect(date,"2022-07-27"))

data_vis_norm <- data_vis %>% 
  filter(date == "2022-07-27") %>% 
  mutate(tank = str_extract(replicate, "C_[1,2,3]|ZF_[1,2,3]|ZL_[1,2,3]|ZH_[1,2,3]|RL_[1,2,3]|RH_[1,2,3]|LL_[1,2,3]|LH_[1,2,3]")) %>% 
  mutate(tank = str_remove(tank, "_")) %>% 
  left_join(pcpn, by = "tank") %>% 
  select(-c("date.y")) %>% 
  rename(date = date.x) %>% 
  mutate(pc_nm = pc_um * 1000) %>% 
  mutate(pn_nm = pn_um * 1000) %>% 
  mutate(nm_metab_per_pc_nm = nM / pc_nm) %>% 
  na.omit() %>% 
  rename(nmol = nM) %>% 
  select(metabolite, date, treatment, nmol, replicate, pn_nm, pc_nm, cn_ratio, nm_metab_per_pc_nm)

write.csv(data_vis_norm, "peridice_metab_data.csv")
```

# STATS
```{r}
# library
library(ggplot2)

# potential y values
y_val <- seq(0, 1, by = 0.01)

# plug in numbers of x and sigma? not sure how to do it theoretically
x <- 0.4
sigma <- 0.2

# dataframe
cdf_data <- data.frame(y = y_values, F_Y_X = pmin(pmax(y_val, x), 1) - x)

# plot
ggplot(cdf_data, aes(x = y, y = F_Y_X)) +
  geom_line() +
  labs(x = "y", y = "F_Y|X(y | x)") +
  ggtitle("Conditional CDF F_Y|X(y | x)") +
  theme_bw()

```



